lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -681,24 +732,55 @@ monitorStakePools tr gp nl DBLayer{..} =
 
 -- | Worker thread that monitors pool metadata and syncs it to the database.
 monitorMetadata
-    :: Tracer IO StakePoolLog
+    :: TVar PoolMetadataGCStatus
+    -> Tracer IO StakePoolLog
     -> GenesisParameters
     -> DBLayer IO
     -> IO ()
-monitorMetadata tr gp DBLayer{..} = do
+monitorMetadata gcStatus tr gp db@(DBLayer{..}) = do
     settings <- atomically readSettings
     manager <- newManager defaultManagerSettings
+
     let fetcher fetchStrategies = fetchFromRemote trFetch fetchStrategies manager
-    forever $ do
-        (refs, successes) <- case poolMetadataSource settings of
-            FetchNone -> pure ([], [])
-            FetchDirect -> fetchThem $ fetcher [identityUrlBuilder]
-            FetchSMASH (unSmashServer -> uri) -> fetchThem $ fetcher [registryUrlBuilder uri]
-
-        when (null refs || null successes) $ do
-            traceWith tr $ MsgFetchTakeBreak blockFrequency
-            threadDelay blockFrequency
+        loop getPoolMetadata = forever $ do
+            (refs, successes) <- getPoolMetadata
+            when (null refs || null successes) $ do
+                traceWith tr $ MsgFetchTakeBreak blockFrequency
+                threadDelay blockFrequency
+
+    case poolMetadataSource settings of
+        FetchNone -> do
+            STM.atomically $ writeTVar gcStatus NotApplicable
+            loop (pure ([], [])) -- TODO: exit loop?
+        FetchDirect -> do
+            STM.atomically $ writeTVar gcStatus NotApplicable
+            loop (fetchThem $ fetcher [identityUrlBuilder])
+        FetchSMASH (unSmashServer -> uri) -> do
+            STM.atomically $ writeTVar gcStatus NotStarted
+            -- health check
+            _ <- healthCheck trFetch (toHealthCheckURI uri) manager
```

Comment:
This does nothing but logging something at the moment, right? May we have a comment explaining the rationale here (rather than repeating the name of the function :sweat_smile:) ? I assume we want to handle the result of the check properly at a later stage and reject client queries which provide a bad URL; it'd be good that a comment explains this rationale, leave a TODO mark, and links to the corresponding ticket.

Reply:
This is removed now